client/src/components/Model.js

```diff
@@ -513,7 +517,7 @@ export default class Model {
       .filter(
         obj =>
           obj.assessment.__recurrence === recurrence &&
-          moment(obj.assessment.__periodStart).isSame(period.start)
+          dateBelongsToPeriod(obj.assessment.__periodStart, period)
```

Comment:
Make it more robust against date-time errors/problems. If we only checked for start date being equal, users would never see anything if not equal. This way, at least we preserve the assessments visibility in the UI.

Reply:
lgtm